Dip/Dip/Dip.swift

```diff
@@ -42,8 +42,7 @@ public class DependencyContainer {
     case Int(IntegerLiteralType)
   }
   
-  private var dependencies = [DefinitionKey : Definition]()
-  private var lock: OSSpinLock = OS_SPINLOCK_INIT
```

Comment:
If we have a lock inside `resolve` method then it will deadlock if you try to call `resolve` inside factory closure or `resolveDependenciesBlock` of definition.